### PR TITLE
fix: Use correct secret, remove gitlab refs

### PR DESCRIPTION
**Pull request by emgrav**

### CODE REVIEW

**Review by FlixCoder — DISMISSED** [review]
LGTM, was it tested? ^^

File: .github/actions/rust-prepare/action.yml [hunk 1]
```diff
@@ -36,12 +36,6 @@ runs:
         GITLAB_SSH: ${{ inputs.gitlab_ssh }}
```
Comment (skomski) [inline .github/actions/rust-prepare/action.yml:36]: This needs to be changed to `CI_SSH_PRIVATE_KEY: "${{ secrets.CI_SSH_PRIVATE_KEY }}"`

**Comment by FlixCoder** [comment]
Do we have a restricted dependency for testing in this repo? ^^

File: .github/actions/rust-prepare/action.yml [hunk 2]
```diff
@@ -5,18 +5,13 @@ inputs:
...
-  gitlab_user:
-    description: "Gitlab username"
-    required: true
-  gitlab_pass:
-    description: "Gitlab username"
-    required: true
   cargo_home:
     description: "Cargo home"
     default: ".cargo"
+    required: true
```
Comment (FlixCoder) [inline .github/actions/rust-prepare/action.yml:11]: why cargo_home required?

File: .github/workflows/rust-workflow.yml [hunk 3]
```diff
@@ -31,9 +31,7 @@ jobs:
 
     - uses: famedly/backend-build-workflows/.github/actions/rust-prepare@main
       with:
-        gitlab_ssh: ${{ secrets.CI_SSH_PRIVATE_KEY}}
-        gitlab_user: ${{ secrets.GITLAB_USER }}
-        gitlab_pass: ${{ secrets.GITLAB_PASS }}
+          gitlab_ssh: ${{ secrets.CI_SSH_PRIVATE_KEY}}
```
Comment (FlixCoder) [inline .github/workflows/rust-workflow.yml:34]: This is somehow indented one too much now I think

File: .github/actions/rust-prepare/action.yml (hunk 2 above)
Comment (skomski) [reply .github/actions/rust-prepare/action.yml:11]: `prepare_rust.sh` creates and puts stuff in the cargo home directory. Most
likely there was a good reason to allow a different name for it but I do not
know it.

Comment (FlixCoder) [reply .github/actions/rust-prepare/action.yml:11]: yeah but the default value applies if not set, why does it need to be required? ^^

The reason it exists is btw mostly gitlab ci / our old runners I think

Comment (skomski) [reply .github/actions/rust-prepare/action.yml:11]: ah thanks sry that was a mistake

**Comment by FlixCoder** [comment]
The tests are still failing with no access to famedly config, so something it still off here ^^

**Comment by skomski** [comment]
> The tests are still failing with no access to famedly config, so something it still off here ^^

yes sorry I thought it will only work on main because the workflow uses `famedly/backend-build-workflows/.github/actions/rust-prepare@main`. But I looked into it and I think the problem is rather that this repo does not have access to `secrets.CI_SSH_PRIVATE_KEY` because it never really worked in the past https://github.com/famedly/backend-build-workflows/actions/runs/6297029018/job/17093178697#step:5:22

**Comment by FlixCoder** [comment]
> > The tests are still failing with no access to famedly config, so something it still off here ^^
> 
> yes sorry I thought it will only work on main because the workflow uses `famedly/backend-build-workflows/.github/actions/rust-prepare@main`. But I looked into it and I think the problem is rather that this repo does not have access to `secrets.CI_SSH_PRIVATE_KEY` because it never really worked in the past https://github.com/famedly/backend-build-workflows/actions/runs/6297029018/job/17093178697#step:5:22

Huh, interesting, can you take a look at it @emgrav ? ^^

**Comment by emgrav** [comment]
Right, I didn't think of this - it actually doesn't make sense to test with private dependencies in this repo as it's public and should not have access to this secret. The reusable workflow will still work when being called from other private repositories as those will have access to the secret and the workflow will inherit it when called.

**Comment by skomski** [comment]
The problem is that the test https://github.com/famedly/backend-build-workflows/actions/runs/6968927714/job/18963795822 cannot succeed without the ssh keys and in the past only worked because the script did not fail on an error. I need to find a solution for that ^^

**Comment by FlixCoder** [comment]
We should probably progress here @emgrav @skomski :D

**Review by FlixCoder — APPROVED** [review]

**Comment by emgrav** [comment]
👀

**Comment by emgrav** [comment]
I don't even remember what this was about, we should probably close it

**Pull request closed without merging**